mala/network/trainer.py

```diff
@@ -912,327 +1010,6 @@ def __process_mini_batch(self, network, input_data, target_data):
-                                                self.static_target_validation,
-                                            )
-                                        else:
-                                            self.static_loss_validation = network.calculate_loss(
-                                                self.static_prediction_validation,
-                                                self.static_target_validation,
-                                            )
-
-                            if self.validation_graph:
-                                self.static_input_validation.copy_(x)
-                                self.static_target_validation.copy_(y)
-                                self.validation_graph.replay()
-                                validation_loss_sum += (
-                                    self.static_loss_validation
-                                )
-                            else:
-                                with torch.cuda.amp.autocast(
-                                    enabled=self.parameters.use_mixed_precision
-                                ):
-                                    prediction = network(x)
-                                    if self.parameters_full.use_ddp:
-                                        loss = network.module.calculate_loss(
-                                            prediction, y
-                                        )
-                                    else:
-                                        loss = network.calculate_loss(
-                                            prediction, y
-                                        )
-                                    validation_loss_sum += loss
-                            if (
-                                batchid != 0
-                                and (batchid + 1) % report_freq == 0
-                            ):
-                                torch.cuda.synchronize(
-                                    self.parameters._configuration["device"]
-                                )
-                                sample_time = time.time() - tsample
-                                avg_sample_time = sample_time / report_freq
-                                avg_sample_tput = (
-                                    report_freq * x.shape[0] / sample_time
-                                )
-                                printout(
-                                    f"batch {batchid + 1}, "  # /{total_samples}, "
-                                    f"validation avg time: {avg_sample_time} "
-                                    f"validation avg throughput: {avg_sample_tput}",
-                                    min_verbosity=2,
-                                )
-                                tsample = time.time()
-                            batchid += 1
-                    torch.cuda.synchronize(
-                        self.parameters._configuration["device"]
-                    )
```

Comment:
I may be missing something, but I cannot find this portion of the code in the new __validate_network - is there a reason why we would want to get rid of the validation graphs? I thought they were working nicely.

Reply:
The computation graphs in validate_network only feed the inputs forward through the network, and later invoke calculate_loss on the model which is just the mean square error. No gradients are accumulated or weights updated. I could potentially leave the computation graphs here just for the LDOS metric, but then all other metrics' calculations would still run outside the computation graph, since it's outside of torch. I could re-use the network predictions out of the graph, but I doubt there is any significant speed improvement compared to evaluation in eager mode. I even doubt whether there is any noticeable improvement when you add the MSE on top. That's why I removed them altogether here. If you strongly feel that this hurts performance when using just the LDOS validation metric or done a benchmark that shows the performance difference I can put it back here.

Reply:
Alright, got it. I think I didn't look into the actual functionality deep enough, just noticed that code that had been authored by Josh had been deleted and just wanted to know if it was by accident or intentional. Your explanation makes sense, I am OK with deleting this part.